ddtrace/contrib/sanic/patch.py

```diff
@@ -258,7 +257,7 @@ async def sanic_http_lifecycle_exception(request, exception):
 
     # Do not attach exception traceback on 404 errors
     # DEV: We still need to set `__dd_span_call_finish` below
-    if not isinstance(exception, NotFound):
+    if not hasattr(exception, "status_code") or 500 <= exception.status_code < 600:
```

Comment:
we probably want to find a way to do this similar to `update_span` which uses `set_http_meta`:

https://github.com/DataDog/dd-trace-py/blob/6d2f07adebec184e98a21a6dcb820284170a04a3/ddtrace/contrib/trace_utils.py#L275-L283

We have per-integration config to define which spans/status codes mark a span as error or not.

Reply:
Thanks for the review! I see. I'm not very familiar with the internals. Do you know if we can just check `span.error` here?

Reply:
I'd need to review this integration again. I cannot remember the lifecycle flow of exceptions.

I would have expected our integration to pick up on the response status code and adjust the error already 🤔 

Although, if it gets marked as an error from being an exception, but then the status code isn't classified as an exception then we aren't "resetting" the status 🤔

Reply:
> Although, if it gets marked as an error from being an exception, but then the status code isn't classified as an exception then we aren't "resetting" the status 🤔

Yea, I think that's the case from my testing. `span.error` gets set correctly to `0` but then `span.set_exc_info()` gets called here on this lifecycle which changes it to `1`. 

Changing this condition to `if span.error` manages to make all the tests pass at least.

Reply:
Thinking a bit more about it I don't think it makes sense to use `span.error`. I don't think there is a guarantee from Sanic on the order in which the signals will run.

We could replace the check for `if not hasattr(exception, "status_code") or config.http_server.is_error_code(exception.status_code):`. This way we don't need to rely on current value from the span at all.

Reply:
```
if not hasattr(exception, "status_code") or config.http_server.is_error_code(exception.status_code):
```

while I'd love to re-use higher level helpers we have, this seems like a good compromise to me.

Reply:
Awesome, I'll make the change and look into the PR checks. Thanks for your time!